remote/remote_state_s3.go

```diff
@@ -173,20 +212,71 @@ func checkIfVersioningEnabled(s3Client *s3.S3, config *RemoteStateConfigS3, terr
 }
 
 // Create the given S3 bucket and enable versioning for it
-func CreateS3BucketWithVersioning(s3Client *s3.S3, config *RemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
-	if err := CreateS3Bucket(s3Client, config, terragruntOptions); err != nil {
+func CreateS3BucketWithVersioning(s3Client *s3.S3, config *ExtendedRemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
+	if err := CreateS3Bucket(s3Client, &config.remoteStateConfigS3, terragruntOptions); err != nil {
 		return err
 	}
 
-	if err := WaitUntilS3BucketExists(s3Client, config, terragruntOptions); err != nil {
+	if err := WaitUntilS3BucketExists(s3Client, &config.remoteStateConfigS3, terragruntOptions); err != nil {
 		return err
 	}
 
-	if err := EnableVersioningForS3Bucket(s3Client, config, terragruntOptions); err != nil {
+	if err := TagS3Bucket(s3Client, config, terragruntOptions); err != nil {
 		return err
 	}
 
+	if err := EnableVersioningForS3Bucket(s3Client, &config.remoteStateConfigS3, terragruntOptions); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func TagS3Bucket(s3Client *s3.S3, config *ExtendedRemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
+
+	if len(config.S3BucketTags) == 0 {
+		terragruntOptions.Logger.Print("No tags for S3 bucket given.")
+		return nil
+	}
+
+	// There must be one entry in the list
+	var tags = config.S3BucketTags[0]
```

Comment:
Why only `[0]`?

Reply:
S3BucketTags can only be empty, or have one element.

If you add several config blocks like this,

```
    config {
      bucket =...

      s3_bucket_tags {
        owner = "team name"
      }

      s3_bucket_tags {
         service = "Terraform"
      }
    }
```

I'm returning this MultipleTagsDeclarations error (see at bottom of remote_state_s3.go file), because in that case, we would have to merge them together, and make sure tags are not set twice, which I found a bit too much and hence decided to allow only one tags declaration.

In case somebody specifies a tag twice in the same configuration, it seems the last value wins, hence no problem for us.

Reply:
If the syntax is:

```hcl
    config {
      bucket =...

      s3_bucket_tags = {
        owner = "team name"
      }

      s3_bucket_tags = {
         service = "Terraform"
      }
    }
```

Could you make `S3BucketTags` a map rather than a list of maps?